### PR TITLE
【NPU】Support npu kernel for fill_constant op

### PR DESCRIPTION
**Pull request by frankwhzhang**
### PR types

New features
### PR changes

OPs
### Describe

add fill_constant npu

### CODE REVIEW

**Comment by frankwhzhang** [comment]
屏幕快照 2021-03-10 下午3 07 39

File: paddle/fluid/operators/fill_constant_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    runner_bc.Run(stream);
+  }
+};
+}  // namespace operators
+}  // namespace paddle
+
+namespace ops = paddle::operators;
+
+REGISTER_OP_NPU_KERNEL(
+    fill_constant,
```
Comment (xymyeah) [inline paddle/fluid/operators/fill_constant_op_npu.cc:82]: fill_constant 是否需要支持bool类型？

Comment (frankwhzhang) [reply paddle/fluid/operators/fill_constant_op_npu.cc:82]: fixed

File: paddle/fluid/operators/fill_constant_op_npu.cc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    auto shape = GetShape(ctx);
+
+    Tensor tensor_tmp(data_type);
+    tensor_tmp.mutable_data<T>({1}, ctx.GetPlace());
+    TensorFromVector(std::vector<T>{value}, ctx.device_context(), &tensor_tmp);
+
+    out_var->mutable_data<T>(shape, place);
+    auto runner_bc = NpuOpRunner("FillD", {tensor_tmp}, {*out_var},
+                                 {{"dims", framework::vectorize(shape)}});
+    runner_bc.Run(stream);
```
Comment (zhiqiu) [inline paddle/fluid/operators/fill_constant_op_npu.cc]: Just curious, what is bc in runner_bc for?

**Review by zhiqiu — APPROVED** [review]
LGTM

File: paddle/fluid/operators/fill_constant_op_npu.cc (hunk 2 above)
Comment (frankwhzhang) [reply paddle/fluid/operators/fill_constant_op_npu.cc]: FIXED

**Pull request merged**